### PR TITLE
A big repo cleanup

### PR DESCRIPTION
**Pull request by gijzelaerr**
* Enable docker builds for various targets, including mypy, pycodestyle and sphinx doc build
* Remove old tox and nosetest stuff
* Add helper Makefile
* Remove old vagrant files
* Formatting fixing
* Depricate mock module, use unitest.mock

### CODE REVIEW

File: snap7/__init__.py [hunk 1]
```diff
@@ -3,14 +3,13 @@
 """
 import pkg_resources
 
-import snap7.server as server
 import snap7.client as client
```
Comment (swamper123) [inline snap7/__init__.py:6]: I think there is the async client missing here.

File: snap7/server.py [hunk 2]
```diff
@@ -287,3 +289,25 @@ def clear_events(self):
         """
         logger.debug("clearing event queue")
         return self.library.Srv_ClearEvents(self.pointer)
+
+
+def mainloop(tcpport: int = 1102):
```
Comment (swamper123) [inline snap7/server.py:294]: May be a stupid question, but why is the mainloop running with port 1102? IsnÄt 102 the S7comm used port?

File: snap7/__init__.py (hunk 1 above)
Comment (gijzelaerr) [reply snap7/__init__.py:6]: it should first be a bit more mature

File: snap7/server.py (hunk 2 above)
Comment (gijzelaerr) [reply snap7/server.py:294]: because then you dont need to run the test suite as root.

**Comment by swamper123** [comment]
Did the Tests even start running? I can't see a result like [here](https://travis-ci.org/github/gijzelaerr/python-snap7/jobs/712819106) and other cases.

**Comment by gijzelaerr** [comment]
there are 3 travis builds that have a NOSCRIPT environment variable, the script.sh bash script checks if that is set and then doesn't execute `make test`

**Comment by gijzelaerr** [comment]
so the actions are actually performed in the install script, where a docker build is triggered. you can also do that yourself using makefile

**Pull request merged**